web-api-validation-expression.md

```diff
@@ -1,98 +1,285 @@
+
+#### Basic Examples
+
+##### Arithmetic Operators
 `3 + 5`
 
 `1 * 3 + 2 - 5`
 
-### Bracketed RETSNAMEs  
-`[ListPrice] > 0`
+##### FIELD_NAME
+To get current value of any field, use the Data Dictionary name: `ListPrice`
 
-### RCP61 added support for unbracketed RETSNAMEs. This also works for DICTNAMEs.
-`ListPrice > 0`
+##### Bracketed FIELD_NAMEs  
+Supported for backwards compatibility with RETS: 
+`[ListPrice]`
 
-### To get current value of any field, just use the `DICTNAME` of that field  
-`ListPrice`
+##### Simple Comparisons
+`ListPrice > 0`
 
-### Access the last value of a field
+##### Access the Last Value of a Field
+`LAST Status != 'Sold'`
 
-`LAST Status != "Sold"`
+##### Field Change Detection
+Returns true if ListPrice has changed.
 
-### Field change detection
 `ListPrice != LAST ListPrice`
 
-### Function calls are also supported
+##### Function Calls
 `foo()`  
-`foo ('bar', 'baz', 42)`
+`foo('bar', 'baz', 42)`
 
-_Cannot use any RETSNAMEs, DICTNAMEs, or reserved words_
+_Functions MUST NOT use any FIELD_NAMEs or reserved words._
 
-### Lists of expressions
+##### Lists of Expressions
 `(ListPrice, Status, 3+5, LAST Status, (1, 2, 3))`
 
-### Supports complex expressions with grouping  
-`ListPrice > 5.01 .AND. (1, 2, 3) .CONTAINS. 3 .OR. (Status .IN. ('Active', 'Pending') .AND. .USERLEVEL. != "Admin")`
+##### Complex Expressions with Grouping  
+`ListPrice > 5.01 .AND. (1, 2, 3) .CONTAINS. 3 .OR. (Status .IN. ('Active', 'Pending') .AND. .MEMBER_MLS_SECURITY_CLASS. != 'Admin')`
 
-### Which parses differently than  
-`ListPrice > 5.01 .AND. (1, 2, 3) .CONTAINS. 3 .OR. Status .IN. ('Active', 'Pending') .AND. .USERLEVEL. != "Admin"`
+##### Which parses differently than  
+`ListPrice > 5.01 .AND. (1, 2, 3) .CONTAINS. 3 .OR. Status .IN. ('Active', 'Pending') .AND. .MEMBER_MLS_SECURITY_CLASS. != 'Admin'`
     
-<br />
-
-## Collection Support
-RCP-019.1 added support for collections, which include `LIST()` and `SET()`. 
+##### Collection Support
+This proposal adds support for `LIST()` and `SET()`.
 
 These functions take 0 or more items.
 
-Backwards compatibility with the previous list syntax of `()` in RCP-019 has been preserved.
-
 | Expression  | Result |  Comments |
 | :--- | :--- | :--- |
-|`LIST(1, 2, 2, 3)`|`LIST(1, 2, 2, 3`| Duplicate items are _preserved_.|
-|`SET(1, 2, 2, 3)`|`SET(1, 2, 3)`| Duplicate items are _removed_.|
+|`LIST(1, 2, 2, 3)`|`LIST(1, 2, 2, 3`| Duplicate items are _preserved_|
+|`SET(1, 2, 2, 3)`|`SET(1, 2, 3)`| Duplicate items are _removed_|
 
 <br />
 
-## DIFFERENCE(), INTERSECTION(), and UNION()  
-RCP-019.1 also added support for `DIFFERENCE()`, `INTERSECTION()`, and `UNION()`,
-which are intended to produce types in the following manner:
+##### List and Set Difference, Intersection, and Union
+This proposal also adds support for the following collection operators:
+* `DIFFERENCE()`
+* `INTERSECTION()`
+* `UNION()`
+

```

Comment:
I don't oppose to have a section here that expresses the purpose of these.

I do think `LIST`, `SET`, `DIFFERENCE`, `INTERSECTION`, and `UNION` should be included in the complete list of standard functions that should be explicitly included in this spec.

Reply:
These are functions that can only operate on `collectionExp`, therefore they are defined alongside it.